### PR TITLE
Add TLS bundle

### PR DESCRIPTION
**Pull request by WRFitch**
## Proposal
Adds a new TLS bundle with the certificate operator added. Also, I removed everything relevant to charm versions, since they aren't used in bundles. 

## TODO 
- Document that the user should upload their own certs for additional security

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -17,25 +16,27 @@ jobs:
       - integration-test
     env:
       CHARMCRAFT_AUTH: ${{ secrets.CHARMHUB_TOKEN }}
+    strategy:
+      matrix:
+        bundle-name: [postgresql-k8s-bundle]
```
Comment (marceloneppel) [inline .github/workflows/release.yaml:21]: We can probably remove the matrix now as there should be only one bundle (with TLS enabled).

**Review by marceloneppel — APPROVED** [review]
LGTM!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+<!-- What does this PR do? -->
+
+## Context
+<!-- Necessary details to understand the proposed changes. -->
+
+## Release Notes
+<!-- A simple bullet-point summary of the changes in this PR. -->
+
+## Testing
+<!-- A summary of how this PR has been tested, including automated testing. -->
```
Comment (MiaAltieri) [inline .github/pull_request_template.md:11]: love this

File: tests/integration/bundle/test_bundle.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    async with ops_test.fast_forward():
+        # Delete the primary pod.
+        model = ops_test.model.info
+        client = AsyncClient(namespace=model.name)
+        await client.delete(Pod, name=primary.replace("/", "-"))
+        logger.info(f"deleted pod {primary}")
+        await ops_test.model.wait_for_idle(
+            apps=[PG, PGB, FINOS_WALTZ], status="active", timeout=600
+        )
```
Comment (MiaAltieri) [inline tests/integration/bundle/test_bundle.py:117]: why not just remove the unit?

**Review by MiaAltieri — COMMENTED** [review]
Looks AWESOME! Minor comments only

File: README.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
 # postgresql-k8s-bundle
 
 A repo containing Juju bundles for PostgreSQL and PGBouncer on k8s.
+
+NB: though version numbers are detailed in the bundle.yaml, they're not used by charmhub. Charmhub only makes the most recent versions on a branch available to download, so this charm will only use the most recent release of each charm.
```
[the remark this answers is not in the document]
Comment (WRFitch) [reply README.md:5]: NB stands for Nota Bene, which is a latin phrase meaning "Note Well". It's common in the UK, but thinking about it I have no idea why we don't just write "note:".

File: tests/integration/bundle/test_bundle.py (hunk 3 above)
Comment (WRFitch) [reply tests/integration/bundle/test_bundle.py]: K8s models don't support removing named units, so to remove the leader unit we need to use this lightkube workaround

File: tests/integration/helpers/postgresql_helpers.py [hunk 5]
```diff
@@ -125,3 +169,36 @@ async def get_unit_address(ops_test: OpsTest, unit_name: str) -> str:
...
+    Returns:
+        the command output if it succeeds, otherwise raises an exception.
+    """
+    complete_command = f"ssh --container postgresql {unit_name} {command}"
+    return_code, stdout, _ = await ops_test.juju(*complete_command.split())
+    if return_code != 0:
+        raise Exception(
+            "Expected command %s to succeed instead it failed: %s", command, return_code
+        )
+    return stdout
```
[the remark this answers is not in the document]
Comment (marceloneppel) [reply tests/integration/helpers/postgresql_helpers.py:191]: Thanks for the information, Mia! In this case the logs are not too big, but we'll take care and use SCP when we have more data to trasnfer.

File: tests/integration/bundle/test_bundle.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+async def test_discover_dbs(ops_test: OpsTest):
+    """Check that proxy discovers new members when scaling up postgres charm.
+
+    Since there's only one readonly endpoint, we can only really test this by querying whether the
+    endpoint exists or not depending on the number of postgres replicas.
+    """
+    pgb_unit = f"{PGB}/0"
+    backend_relation = get_backend_relation(ops_test)
+    backend_databag = await get_app_relation_databag(ops_test, pgb_unit, backend_relation.id)
+    assert not backend_databag.get("read-only-endpoints", None)
```
[the remark this answers is not in the document]
Comment (marceloneppel) [reply tests/integration/bundle/test_bundle.py]: This read only endpoint value (when present) always points to the read only PostgreSQL instances (it uses a K8S services that point). There is already a check in the PostgreSQL charm repo, so probably it's optional here. The idea here is to check just that initially there were no replicas and after scaling we have new instances (the replicas).

File: tests/integration/bundle/test_bundle.py [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    await scale_application(ops_test, PGB, 3)
+    await ops_test.model.wait_for_idle()
+    await scale_application(ops_test, PG, 3)
+
+    updated_backend_databag = await get_app_relation_databag(
+        ops_test, pgb_unit, backend_relation.id
+    )
+    assert updated_backend_databag.get(
+        "read-only-endpoints", None
+    ), f"read-only-endpoints not populated in updated backend databag - {updated_backend_databag}"
```
[the remark this answers is not in the document]
Comment (marceloneppel) [reply tests/integration/bundle/test_bundle.py]: I added the explanation in the previous comment. Please let me know if it makes sense.

**Review by MiaAltieri — APPROVED** [review]

**Pull request merged**